File: ruby/clisoap1.rb
require 'webrick'
require 'savon'

server = WEBrick::HTTPServer.new(
  Port: 8000
)

server.mount_proc '/' do |req, res|

  numero = req.query['n']

  client = Savon.client(
    wsdl: 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL'
  )

  response = client.call(
    :number_to_words,
    message: { ubiNum: numero }
  )

  resultado =
    response.body[:number_to_words_response]
            [:number_to_words_result]

  res.body = resultado

end

trap('INT') { server.shutdown }

server.start

File: perl/clisoap1.pl
use strict;
use warnings;
use SOAP::Lite;
use HTTP::Server::Simple::CGI;

{
    package MyServer;
    use base 'HTTP::Server::Simple::CGI';

    sub handle_request {
        my ($self, $cgi) = @_;

        my $numero = $cgi->param('n');

        my $soap = SOAP::Lite
            ->service(
            'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL'
        );

        my $resultado =
            $soap->NumberToWords($numero);

        print "HTTP/1.0 200 OK\r\n";
        print "Content-Type: text/plain\r\n\r\n";
        print $resultado;
    }
}

my $server = MyServer->new(8000);
$server->run();